### DISCUSSION TITLE
Porque o mercado para quem trabalha com React  é tão pequeno no Brasil?

### DISCUSSION THREAD

**Discussion by marcusvcribeiro** [frontendbr/forum#767] [Discussão]
Algum tempo atrás venho observando isso, a quantidade de vagas de angular são bem maiores que a quantidade de vagas para quem trabalhar com React no Brasil, será porque isso?

**Comment by yanmagale** [comment]
@marcusvcribeiro acredito que existem uma série de fatores para poder se explicar este fato.  Talvez uma das coisas que possam explicar esta diferença seja o *tempo* de existência destas tecnologias.

O Angular já esta no mercado desde 2010, é uma tecnologia já bem difundida, com diversas iniciativas, cursos e conteúdos....existem também muitas talks e empresas utilizando. Por consequência, é muito natural ter muitas vagas pedindo esta tecnologia. O React por sua vez já bem mais novo(~2014), mas que cresceu muito nos últimos anos. A comunidade tem aumentado bastante e cada vez mais vejo vagas de empresas utilizando React em seus projetos (aqui no Front-End BR mesmo sempre tem uma vaga pedindo React), o ecossistema a volta do React tem crescido muito, então a tendência de termos mais vagas em React com certeza irã aumentar. Contudo, com relação a Angular(ou qualquer outra tecnologia que tenha surgido antes), não da para compararmos devido ao tempo de existência + amadurecimento da tecnologia + amadurecimento e aceitação da comunidade ...

**Comment by doug2k1** [comment]
Aqui no https://github.com/frontendbr/vagas atualmente tem 68 vagas abertas com o termo "angular" e 84 com o termo "react". 

Como o @yanmagale disse, Angular tem mais tempo, mas o React vem ganhando força.

**Comment by rubenmarcus** [comment]
Lá fora tem muita vaga de React. Vue então aqui quase não tem. Vamos ajudar a difundir!!

**Comment by felipeuntill** [comment]
Se `Angular` e `React` nada mais são que libs/frameworks `Javascript` porque não estudar e dominar ambos os eco-sistemas? 

É fato que a grande maioria das empresas optam pelo Angular mas o React também não é nenhuma bala de prata, eu vejo muita gente perdendo oportunidade por escolher framework/libs em vez de escolher trabalhar.

_O mercado de React não está pequeno, é front-end e toda empresa precisa de um. Cabe aos profissionais decidirem se querem trabalhar com "React" ou como "Front-Ends"..._

**Comment by fernandofleury** [comment]
Como assim? Praticamente toda vaga de uma área de desenvolvimento "real" pede react hoje em dia.

**Comment by fabriciofmsilva** [comment]
Sobre as vagas, eu penso que frameworks/bibliotecas é meio que moda, cada momento uma vai estar mais popular que a outra. Com isso as vagas também. Outro ponto a se olhar é qual o tipo de vaga que você está procurando. Eu vejo o React sendo mais pedido por empresas mais modernas, como startups, e o Angular em empresas mais engessadas, como bancos e multinacionais.

Voltando ao ponto dos frameworks/bibliotecas, o esquema é sempre estudar os que estão na ponta. Se você trabalha com Angular faça projetos pessoais e open source com o React ou Vue. Pode ser que num momento você encontre uma oportunidade de usar algo diferente onde trabalha. Se não pintar pelo menos você já tem o conhecimento para entender os pontos fortes e fracos de cada "ecossistema".

Não precisa estudar todos os hypes, mas sempre dê uma olhada neles. Daí você se aprofunda mais naqueles que você: ou precisa dele para trabalhar, ou tem afinidade. Eu só deixaria um pouco de mais tempo para estudar JavaScript e DOM, a lá Vanilla, pois no final é tudo JavaScript e manipulação de DOM. ;)

**Comment by felipefialho** [comment]
Acho estranho focar a vaga em um framework especifico. 

Tipo: `React Developer`, mesmo sabendo que eles tem especificidades que pessoas com mais conhecimentos tratam com mais facilidade. 

Sou da opinião que a curva de aprendizado entre os SPAs é pequena, mesmo sendo diferentes. 

Por isso, penso que buscar alguém que já tenha trabalhado com algum SPA é o suficiente.
